frontend/integration-tests/tests/kubevirt/non-admin.scenario.ts

```diff
@@ -0,0 +1,148 @@
+import { KUBEADMIN_IDP, KUBEADMIN_USERNAME, BRIDGE_KUBEADMIN_PASSWORD, BRIDGE_HTPASSWD_IDP, BRIDGE_HTPASSWD_USERNAME, BRIDGE_HTPASSWD_PASSWORD } from './utils/consts';
+import * as loginView from '../../views/login.view';
+import * as crudView from '../../views/crud.view';
+import Wizard from './models/wizard';
+import Yaml from './models/yaml';
+import { logout } from '../../views/kubevirt/login.view';
+import { execSync } from 'child_process';
+
+describe('Test nonadmin user behaviour', () => {
+  const nonAdminNS = `${testName}-nonadmin`;
+
+  const verifyPermissions = async function(ns: string, perm: string) {
+    const wizard = new Wizard();
+    const yaml = new Yaml();
+
+    await browser.get(`${appHost}/k8s/ns/${ns}/virtualmachines`);
+    if (perm === 'noAccess') {
+      await browser.wait(until.presenceOf(crudView.errorMessage));
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await wizard.openWizard();
+      await wizard.close();
+
+      await yaml.openYamlPage();
+      await yaml.cancelCreateVM();
+    }
+
+    await browser.get(`${appHost}/k8s/ns/${ns}/vmtemplates`);
+    if (perm === 'noAccess') {
+      await browser.wait(until.presenceOf(crudView.errorMessage));
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await wizard.openWizard();
+      await wizard.close();
+    }
+
+    await browser.get(`${appHost}/overview/ns/${ns}/`);
+    if (perm === 'noAccess') {
+      await browser.wait(until.presenceOf(crudView.errorMessage));
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+    }
+  };
+
+  const verifyPermissionWithRoleBinding = async function(ns: string, binding: string, role: string, perm: string) {
+    execSync(`oc adm policy add-${binding}-to-user ${role} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+    await verifyPermissions(perm, ns);
+    execSync(`oc adm policy remove-${binding}-from-user ${role} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+  };
+
+  beforeAll(async() => {
+    try {
+      execSync('oc get -o yaml -n openshift-config secret test');
```

Comment:
Maybe we could use more descriptive name of the secret than `test` ?

Reply:
How about `test-secret`?

Reply:
sounds good